ga4gh/datamodel/biodata.py

```diff
@@ -0,0 +1,114 @@
+"""
+Dataset objects
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import datetime
+
+import ga4gh.datamodel as datamodel
+import ga4gh.protocol as protocol
+import ga4gh.exceptions as exceptions
+
+
+class AbstractBioSample(datamodel.DatamodelObject):
+    """
+    This class represents an abstract BioSample object.
+    It sets default values and getters, as well as the
+    toProtocolElement function.
+    """
+    compoundIdClass = datamodel.BioSampleCompoundId
+
+    def __init__(self, parentContainer, localId):
+        super(AbstractBioSample, self).__init__(parentContainer, localId)
+        self._createDateTime = datetime.datetime.now().isoformat()
```

Comment:
The meaning and of the time fields is needs clarified.  Right
now the created/updated fields were intended, but not
documented, as load time.  This is not that useful for
most bioinfo sharing.

There is a ticket that covers this.  This is find for
now until that is resolve.

Maciek Smuga-Otto <notifications@github.com> writes:
> In ga4gh/datamodel/biodata.py:
> 
> > +import ga4gh.datamodel as datamodel
> > +import ga4gh.protocol as protocol
> > +import ga4gh.exceptions as exceptions
> > +
> > +
> > +class AbstractBioSample(datamodel.DatamodelObject):
> > +    """
> > +    This class represents an abstract BioSample object.
> > +    It sets default values and getters, as well as the
> > +    toProtocolElement function.
> > +    """
> > +    compoundIdClass = datamodel.BioSampleCompoundId
> > +
> > +    def __init__(self, parentContainer, localId):
> > +        super(AbstractBioSample, self).__init__(parentContainer, localId)
> > +        self._createDateTime = datetime.datetime.now().isoformat()
> 
> Shouldn't the creation time be sourced from the JSON (indicating, perhaps, when
> the biosample was collected or sequenced?)
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub*
>